### PR TITLE
Add slack notification when user starts a deployment to dev

### PR DESCRIPTION
**Pull request by waldofouche**
When a user triggers a new deployment with `/deploy` it will trigger a new webhook and post to the slack channel. Will remove the need to keep checking if anyone is using x app in dev:

Screenshot 2024-10-28 at 1 07 20 pm

Once merged, will add all tech team users to the channel

### CODE REVIEW

File: .github/workflows/deploy-sync.yml [hunk 1]
```diff
@@ -155,6 +157,11 @@ jobs:
           BRANCH_NAME: ${{ steps.branch.outputs.BRANCH_NAME }}
         id: validate_issue_comment
 
+      - uses: ausaccessfed/workflows/.github/actions/alert_webhook@feature/add-slack-of-dev-deployments
+        with:
+          MESSAGE:MESSAGE: "{\"text\":\"The following user '${{ github.actor }}' has started a dev deployment of '${{ github.repository }}' with commit '${{ github.sha }}\'\"}"
```
Comment (waldofouche) [inline .github/workflows/deploy-sync.yml:162]:
```suggestion
          MESSAGE: "{\"text\":\"The following user '${{ github.actor }}' has started a dev deployment of '${{ github.repository }}' with commit '${{ github.sha }}\'\"}"
```

File: .github/workflows/deploy-sync.yml [hunk 2]
```diff
@@ -155,6 +157,11 @@ jobs:
           BRANCH_NAME: ${{ steps.branch.outputs.BRANCH_NAME }}
         id: validate_issue_comment
 
+      - uses: ausaccessfed/workflows/.github/actions/alert_webhook@feature/add-slack-of-dev-deployments
+        with:
+          MESSAGE: "{\"text\":\"The following user '${{ github.actor }}' has started a dev deployment of '${{ github.repository }}' with commit '${{ github.sha }}\'\"}"
```
Comment (waldofouche) [inline .github/workflows/deploy-sync.yml:162]:
```suggestion
          MESSAGE: "{\"text\":\"The following user '${{ github.actor }}' has started a dev deployment of '${{ github.repository }}' with commit '${{ github.sha }}'\"}"
```

File: .github/actions/alert_webhook/action.yaml [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+  WEBHOOK_URL:
+    description: 'The URL of the webhook to notify'
+    required: true
+runs:
+  using: 'composite'
+  steps:
+    - name: Send notification to webhook
+      shell: bash
+      run: |
+        curl -X POST -H 'Content-type: application/json' --data "{\"text\":\"${{ inputs.message }}\"}" "${{ inputs.webhook_url }}"
```
Comment (matthew-puku) [inline .github/actions/alert_webhook/action.yaml:17]: not all webhooks accept this JSON structure `"{"text": "blah" }"` so I don't think the generic naming is right. The payload is specific to slack webhooks, so the names/descriptions should be specific as well.

**Review by matthew-puku — APPROVED** [review]

**Comment by phyzical** [comment]
@waldofouche ah no what i meant is another flow in summarize please

**Comment by phyzical** [comment]
```
it would be generic like if commit_sha do it, then it just takes commit_status, commit_url, commit_description all optional
commit_status default == success
commit_url default == ''
commit_description default == '' (edited) 
```

**Comment by waldofouche** [comment]
@phyzical yeah I had a look at that one and it was not really relevant here imo as it'll just send a single small message, rather than build out that complex message in the shell script, it might even be possible to use this workflow in that one

File: .github/actions/alert_webhook/action.yaml (hunk 3 above)
Comment (waldofouche) [reply .github/actions/alert_webhook/action.yaml]: Yeah true true, fixed in https://github.com/ausaccessfed/workflows/pull/402/commits/9a74fb163be53a05e817644407b4637b63e5cfe7

**Pull request merged**